File: Problemas/cpp/bancos-con-herencia/c++/cuenta-corriente.cc
#include "cuenta-corriente.h"
#include <cmath>

using namespace std;

CuentaCorriente::CuentaCorriente(double _capital, double _interes)
	: Cuenta(_capital), interes(_interes)
{
}

double CuentaCorriente::valor(int t) const
{
	return capital*pow(1.0 + interes/100.0,t);
}

File: Problemas/cpp/bancos-con-herencia/java/CuentaCorriente.java
public final class CuentaCorriente extends Cuenta
{
	private double interes;

	public CuentaCorriente(double _capital, double _interes)
	{
		super(_capital); //Constructor clase base
		interes = _interes;
	};

	// Implementa el calculo del valor
	@Override
	public double valor(int t)
	{
		return capital*Math.pow(1.0 + interes/100.0,t);
	}
}
